File: 0049_group_anagram/main.py
class Solution:
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        groups = {}
        res = []
        for s in strs:
            key = ''.join(sorted(s))
            if key in groups:
                groups[key].append(s)
            else:
                groups[key] = [s]
        for _, v in groups.items():
            res.append(v)
        return res

File: 0049_group_anagram/main.go
package main

import "sort"

func groupAnagrams(strs []string) [][]string {
	group := map[string][]string{}
	res := [][]string{}
	for _, s := range strs {
		str := []byte(s)
		sort.Slice(str, func(i, j int) bool {
			return str[i] <= str[j]
		})
		key := string(str)
		_, ok := group[key]
		if ok {
			group[key] = append(group[key], s)
		} else {
			group[key] = []string{s}
		}
	}
	for _, v := range group {
		res = append(res, v)
	}
	return res
}

func main() {

}
